### PR TITLE
Add ability to specify Image source with uri and cache

### PR DESCRIPTION
**Pull request by jaischeema**
This will let you use images from local bundled assets, very simple first version, can be extended as per requirements.

Example usage:
```elm
import NativeUi.Elements as Elements exposing (..)
import NativeUi.Image as Image

imageTag : String -> Node msg
imageTag assetName =
    Elements.image
        [ Ui.style
            [ Style.height 32
            , Style.width 32
            ]
        , Image.source (Image.defaultSource assetName)
        ]
        []
```

### CODE REVIEW

File: src/NativeUi/Image.elm [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        )
+
+{-| elm-native-ui Image
+
+@docs Source, CacheStrategy, source
+-}
+
+import Json.Encode as Encode
+import Json.Decode as Decode
+import NativeUi exposing (Property, Node, node, renderProperty, property)
```
Comment (sdaves) [inline src/NativeUi/Image.elm]: Do Node and node need imported?

**Review by sdaves — APPROVED** [review]
Nice addition and what a handy way to handle image caching!

File: src/NativeUi/Image.elm (hunk 1 above)
Comment (jaischeema) [reply src/NativeUi/Image.elm]: No, its not. Will clean it up 👍

File: src/NativeUi/Image.elm [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+
+{-| -}
+type alias Source =
+    { uri : String
+    , cache : Maybe CacheStrategy
+    }
+
+
+defaultSource : String -> Source
```
Comment (ohanhi) [inline src/NativeUi/Image.elm]: missing doc comment

**Comment by ohanhi** [comment]
Great, thanks!

**Pull request merged**